libraries/ESP8266WiFi/src/WiFiClientSecureBearSSL.cpp

```diff
@@ -235,6 +235,28 @@ int WiFiClientSecure::connect(const String& host, uint16_t port) {
   return connect(host.c_str(), port);
 }
 
+int WiFiClientSecure::connectAsync(const char* name, uint16_t port, bool* waiting) {
+  if (waiting != nullptr) *waiting = false;
+  if (connected()) return 1;
```

Comment:
A `connect()` on a previously connected client should drop the current connection and open a new one.  This won't, it will return success.